Add url encode function

larsalbrecht opened the pull request:
If urls of e.g. episodes contains "ß", the php filter_var function returns false.
So I added a function that encode the url before the check (for further errors) and save this new encodedUrl.

I also cleaned up the code a little bit.

bohwaz commented:
Please don't change indentation of changed files, or I won't be able to accept your PR :)

larsalbrecht commented:
> Please don't change indentation of changed files, or I won't be able to accept your PR :)

You are right, that was not good. I reverted the indentation and added an .editorconfig to define the tab for *.php files.

bohwaz reviewed (commented):
Except from what I mentioned in the comments, everything is great, thank you :)

larsalbrecht commented on server/index.php:
```diff
@@ -80,7 +84,7 @@
 }
 else {
-	echo '<p><a href="login">Login</a></p>';
+	echo '<p><a href="/login">Login</a></p>';
```
Done

larsalbrecht commented on server/inc/API.php:
```diff
@@ -6,7 +6,7 @@ class API
 	protected ?string $section;
 	public ?string $url;
-	public ?string $base_url;
+	private ?string $base_url;
```
Done

larsalbrecht commented on server/inc/API.php:
```diff
@@ -32,12 +32,19 @@ public function __construct(DB $db)
-	public function url(string $path = ''): string
+	private function encodeUrl(string $url): string
+	{
+		return preg_replace_callback('#://([^/]+)/([^?]+)#', static function ($match) {
```
done. I add an unparseUrl function to get the correct url back.

larsalbrecht commented:
You can review again. The conservations are open, so you can add additonal comments if you have anything else :)
